### PR TITLE
Add reason testing library documentation

### PR DESCRIPTION
**Pull request by wyze**
It was suggested by Kent at https://github.com/wyze/bs-dom-testing-library/issues/5 that I move over some of the documentation to the website here. I've done more than just the above since I maintain the bindings around `(dom|react)-testing-library` for Reason, as well as `jest-dom` bindings.

Definitely provide feedback on the structure as I wasn't sure the best way to lay it out.

Also, I ran `yarn format-docs` and looks like there were some changes unrelated to what I made. I'll note those below.

### CODE REVIEW

File: docs/api-queries.md [hunk 1]
```diff
@@ -283,8 +283,8 @@ const usernameInputElement = getByTestId(container, 'username-input')
 > case. Using data-testid attributes do not resemble how your software is used
 > and should be avoided if possible. That said, they are _way_ better than
 > querying based on DOM structure or styling css class names. Learn more about
-> `data-testid`s from the blog post ["Making your UI tests resilient to
-> change"](https://blog.kentcdodds.com/making-your-ui-tests-resilient-to-change-d37a6ee37269)
+> `data-testid`s from the blog post
+> ["Making your UI tests resilient to change"](https://blog.kentcdodds.com/making-your-ui-tests-resilient-to-change-d37a6ee37269)
```
Comment (wyze) [inline docs/api-queries.md]: Happened with `yarn format-docs`.

File: docs/example-input-event.md [hunk 2]
```diff
@@ -6,7 +6,9 @@ sidebar_label: Input Event
 
 > **Note**
 >
-> If you want to simulate a more natural typing behaviour while testing your component, consider the companion library [`user-event`](ecosystem-user-event.md)
+> If you want to simulate a more natural typing behaviour while testing your
```
Comment (wyze) [inline docs/example-input-event.md]: Happened with `yarn format-docs`.

File: docs/learning.md [hunk 3]
```diff
@@ -4,7 +4,9 @@ title: Learning Material
 sidebar_label: Learning Material
 ---
 
-- [TestingJavaScript.com](https://testingjavascript.com) A HUGE course all about testing JavaScript applications (See also the [course material with many examples using react-testing-library](https://github.com/kentcdodds/react-testing-library-course))
+- [TestingJavaScript.com](https://testingjavascript.com) A HUGE course all about
```
Comment (wyze) [inline docs/learning.md]: Happened with `yarn format-docs`.

**Comment by alexkrolick** [comment]
Do you also want to add a link + logon the homepage?

File: website/sidebars.json [hunk 4]
```diff
@@ -45,6 +45,17 @@
         "type": "subcategory",
         "label": "Angular Testing Library",
         "ids": ["angular-testing-library/intro"]
+      },
+      {
+        "type": "subcategory",
+        "label": "Reason Testing Libraries",
```
Comment (alexkrolick) [inline website/sidebars.json]: Even though there are technically separate wrappers for dom- and react-, can we use singular (library) in the heading? Or would that be confusing?

Comment (alexkrolick) [reply website/sidebars.json]: Or even rename all the wrapper headings to just "React", "Reason", etc.

Comment (wyze) [reply website/sidebars.json]: Sure. I could do Reason Testing Library and Reason-React Testing Library

**Comment by wyze** [comment]
@alexkrolick Changes have been pushed!

**Comment by alexkrolick** [comment]
Looks good to me!

**Pull request merged**